colorizer_data/writer.py

```diff
@@ -130,22 +140,76 @@ class ColorizerDatasetWriter:
 
     outpath: Union[str, pathlib.Path]
     manifest: DatasetManifest
+    backdrops: Dict[str, BackdropMetadata]
     scale: float
 
     def __init__(
         self,
         output_dir: Union[str, pathlib.Path],
         dataset: str,
         scale: float = 1,
+        force_overwrite: bool = False,
     ):
         self.outpath = os.path.join(output_dir, dataset)
         os.makedirs(self.outpath, exist_ok=True)
         self.scale = scale
-        self.manifest = {"features": []}
 
-    def write_feature(self, data: np.ndarray, info: FeatureInfo):
+        # Check for existence of manifest file in expected path
+        manifest_path = self.outpath + "/manifest.json"
+        if os.path.exists(manifest_path) and not force_overwrite:
+            # Load in the existing file
+            try:
+                with open(manifest_path, "r") as f:
+                    self.manifest = json.load(f)
+                logging.info(
+                    "An existing manifest file was found in the output directory and will be updated."
+                )
+            except:
+                logging.warning(
+                    "A manifest file exists in this output directory but could not be loaded, and will be overwritten instead!"
+                )
+                self.manifest = {}
+        else:
+            self.manifest = {}
+
+        # Clear features
+        self.manifest["features"] = []
+
+        # Load backdrops from existing manifest, if applicable
+        self.backdrops = {}
+        if "backdrops" in self.manifest:
+            for backdrop_metadata in self.manifest["backdrops"]:
+                self.backdrops[backdrop_metadata["key"]] = backdrop_metadata
+
+    def write_categorical_feature(self, data: np.ndarray, info: FeatureInfo) -> None:
```

Comment:
why can't `write_feature` just call this internally?  Then client scripts will not have to write their own logic to distinguish and can still just call write_feature for everything

Reply:
Resolved in #30!